packages/shared/lib/core/wallet/classes/activity.class.ts

```diff
@@ -45,6 +52,41 @@ export class Activity implements IActivity {
         return this
     }
 
+    setFromOutput(
+        outputId: string,
+        output: OutputData,
+        accountAddress: string,
+        hidden: boolean,
+        claimed: boolean
+    ): Activity {
+        const address =
+            output?.address?.type === 0
+                ? Bech32Helper.toBech32(0, Converter.hexToBytes(output.address.pubKeyHash.substring(2)), 'rms')
+                : ''
+        const isIncoming = address === accountAddress
+        const isInternal = !!findAccountWithAddress(address)
+        this.id = outputId
+        this.outputId = outputId
```

Comment:
Why duplicate `id` and `outputId` ?

Reply:
we need some unique identifier in the activities so we can remove/update them in our stores. 
Activities from transactions additionally have `transactionId` saved, so that we e.g. access the transaction in the explorer. The outputId is not used ATM, but i think we will need it for e.g. fetching the transaction resulting in that output